### PR TITLE
signIn and signUp working

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
signin
signUp

files to look for:
signIn.jsx
signUp.jsx
Nav.jsx

### CODE REVIEW

File: src/context/Authentication.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createContext, useContext, useState } from "react"
+
+
+const AuthContext=createContext();
+
+export function AuthProvider({children}){
+    const [login,setLogin]=useState(false);
+    // const [loginBtn,setLoginBtn]=useState("Log In");
```
Comment (gauravsinhaweb) [inline src/context/Authentication.jsx:8]: remove comments

File: src/pages/Nav/Nav.jsx [hunk 2]
```diff
@@ -1,7 +1,23 @@
 import "./nav.css";
 import * as All from "../../assets/icons/icons.jsx"
-import {Route,Link,Routes} from "react-router-dom"
+import {Route,Link,Routes, useLocation} from "react-router-dom"
+import { useAuth } from "../../context/Authentication";
+import { useState } from "react/cjs/react.production.min";
 export function Nav(){
+    const {login,setLogin}=useAuth();
+    const location=useLocation();
+    console.log(location.pathname)
```
Comment (gauravsinhaweb) [inline src/pages/Nav/Nav.jsx:9]: remove all the console logs before commit

File: src/pages/Sign In/signIn.css [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    column-gap: 1rem;
+    row-gap: 9px;
+}
+
+.form-label{
+    color: var(--color-white);
+}
+
+
+
```
Comment (gauravsinhaweb) [inline src/pages/Sign In/signIn.css:17]: remove extra spacing

File: src/pages/Sign In/signIn.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import "./signIn.css"
+import {Link} from "react-router-dom"
+
```
Comment (gauravsinhaweb) [inline src/pages/Sign In/signIn.jsx:3]: remove extra spacing

**Review by gauravsinhaweb — APPROVED** [review]
Good to merge after changes, good use of semantic HTML

File: src/pages/Sign In/signIn.jsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import axios from "axios";
+import { useState } from "react/cjs/react.development";
+
+export function SignIn(){
+    const {login,setLogin}=useAuth();
+    const [loginData,setLoginData]=useState({"email":"","password":""});
+    
+    const onChangeHandler=(e)=>{
+        return setLoginData(loginData=>({...loginData,[e.target.name]:e.target.value}))
+    };
```
Comment (m26unkwn) [inline src/pages/Sign In/signIn.jsx:14]: Instead returning you can put setLoginData(..)

File: src/pages/Sign Up/signup.jsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import { useState } from "react";
+import {Link} from "react-router-dom";
+
+import axios from "axios";
+import { useAuth } from "../../context/Authentication";
+import { useReducer } from "react/cjs/react.development";
+
+export function SignUp(){
+   const[passwordType,setPasswordType]=useState("password");
+   const[state,dispatch]=useReducer(details,{firstName:"",lastName:"",email:"",password:"",confirmPassword:""});
```
Comment (m26unkwn) [inline src/pages/Sign Up/signup.jsx:13]: you can use detailReducer instead of `details` for good naming structure

**Review by m26unkwn — APPROVED** [review]

File: src/pages/Sign In/signIn.jsx (hunk 5 above)
Comment (dev-ashishsethi) [reply src/pages/Sign In/signIn.jsx:14]: will that not result in infinite loop?

File: src/pages/Sign Up/signup.jsx (hunk 6 above)
Comment (dev-ashishsethi) [reply src/pages/Sign Up/signup.jsx:13]: thanks for suggestion

**Pull request merged**